File: rsa/fermat's_factorization.js
function fermat(n) {
    let x = Math.ceil(Math.sqrt(n))
    if (x ** 2 == n) {
        return [x, x]
    }
    for (let i=0;i<50;i++){
        const l=x**2-n
        const y=Math.sqrt(l)
        if(y % 1 === 0){
            return [x-y,x+y]
        }
        x++
    }
}

File: rsa/fermat's_factorization.py
import math
from decimal import *
getcontext().prec = 250
def fermat(n):
    x=Decimal(n).sqrt().__ceil__()
    if x**2==n:
        return [x,x]
    for i in range(80000):
        print(f'iteration {i}')
        l=x**2-n
        y=Decimal(l).sqrt()
        if(y%1==0):
            return [x-y,x+y]
        x+=1
print(fermat(42349*48527))

